crates/wasi-common/src/sys/unix/bsd/hostcalls_impl.rs

```diff
@@ -57,25 +59,30 @@ pub(crate) fn path_symlink(old_path: &str, resolved: PathGet) -> Result<()> {
     log::debug!("path_symlink old_path = {:?}", old_path);
     log::debug!("path_symlink resolved = {:?}", resolved);
 
+    eprintln!("path_symlink: resolved.path='{:?}'", resolved.path());
     unsafe { symlinkat(old_path, resolved.dirfd().as_raw_fd(), resolved.path()) }.or_else(|err| {
+        eprintln!("  path_symlink: error: {:?}", err);
         if let YanixError::Errno(errno) = err {
+            eprintln!("    path_symlink: yanix error: {:?}", errno);
             match errno {
                 Errno::ENOTDIR => {
+                    eprintln!("      path_symlink: yanix ENOTDIR");
                     // On BSD, symlinkat returns ENOTDIR when it should in fact
                     // return a EEXIST. It seems that it gets confused with by
                     // the trailing slash in the target path. Thus, we strip
                     // the trailing slash and check if the path exists, and
                     // adjust the error code appropriately.
-                    let new_path = resolved.path().trim_end_matches('/');
-                    if let Ok(_) = unsafe {
-                        fstatat(
-                            resolved.dirfd().as_raw_fd(),
-                            new_path,
-                            AtFlag::SYMLINK_NOFOLLOW,
-                        )
-                    } {
+                    let dirfd = resolved.dirfd().as_raw_fd();
+                    let new_path = trim_end_slashes(resolved.into_path());
```

Comment:
I believe this line is generating an error on Mac. Since `into_path` moves out of `PathGet` struct, we drop the `dirfd` which is of type `File` and not `ManuallyDrop` causing a "bad file descriptor" error. I'll provide a fix by swapping `into_path` into `path` and then cloning the path instead. If we want to avoid unnecessary deep copies, an alternative would be a method on `PathGet::into_components` or something which moves *all* fields out of the struct into a tuple. Then we'll preserve the lifetime of `dirfd` just long enough for this to work without a deep copy.